File: inference_framework-java/archive/python_original/inference_framework/models/base_model.py
"""Base implementation of ModelInterface."""
from inference_framework.model_interface import ModelInterface
from inference_framework.output_event import OutputEvent
from typing import Dict, Any, List
import polars as pl
import logging

logger = logging.getLogger(__name__)

class BaseModel(ModelInterface):
    """Base model with common utilities."""
    
    def __init__(self, config: Dict[str, Any] = None):
        self.config = config or {}
        self.model_id = self.__class__.__name__.lower()
    
    def process(self, df: pl.DataFrame, features: Dict[str, Dict[str, Any]]) -> List[OutputEvent]:
        """Default process - should be overridden."""
        logger.warning(f"BaseModel process called - override in subclass")
        return []

File: inference_framework-java/src/main/java/com/inferenceframework/model/models/BaseModel.java
package com.inferenceframework.model.models;  // model sub-module

import com.inferenceframework.event.OutputEvent;
import com.inferenceframework.model.ModelInterface;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

/**
 * Base implementation of ModelInterface.
 * Adapted for Java: uses List<Map<String, Object>> instead of Polars DataFrame.
 */
public abstract class BaseModel implements ModelInterface {
    protected static final Logger logger = LoggerFactory.getLogger(BaseModel.class);

    protected Map<String, Object> config;
    protected String modelId;

    /**
     * Constructor.
     */
    public BaseModel(Map<String, Object> config) {
        this.config = config != null ? config : new HashMap<>();
        this.modelId = this.getClass().getSimpleName().toLowerCase();
    }

    /**
     * Default process - should be overridden by subclasses.
     * In base, logs warning and returns empty list.
     */
    @Override
    public List<OutputEvent> process(List<Map<String, Object>> messages, Map<String, Map<String, Object>> features) {
        logger.warn("BaseModel process called - override in subclass");
        return new ArrayList<>();
    }
}